src/lib.rs

```diff
@@ -1539,7 +1544,10 @@ impl GooseAttack {
             users.push(user);
             self.stats.users += 1;
             debug!("sleeping {:?} milliseconds...", sleep_duration);
-            tokio::time::delay_for(sleep_duration).await;
+            tokio::time::delay_for(sleep_duration - user_thread_started.elapsed()).await;
+
+            // Update user_thread_started after each delay.
+            user_thread_started = time::Instant::now();
```

Comment:
The helper can encapsulate this update, too.

Reply:
implemented